### PR TITLE
Storage Selector implementation

### PR DESCRIPTION
**Pull request by alex-deaconu**
New CH version, storage constants
Directive replaces old one

Add Folder modal
File upload
Subscription status integration

Split out storageFactory for constants
Reset folderPath every time Storage is opened

### CODE REVIEW

File: web/scripts/storage/controllers/ctr-storage-selector-modal.js [hunk 1]
```diff
@@ -0,0 +1,146 @@
+'use strict';
+angular.module('risevision.storage.controllers')
+  .controller('StorageSelectorModalController', ['$scope',
+    'fileSelectorFactory', 'filesFactory', '$modalInstance', '$loading',
```
Comment (alex-deaconu) [inline web/scripts/storage/controllers/ctr-storage-selector-modal.js]: Will split this controller in 2 pieces for the Full Screen implementation: File List & Selector Modal.

File: web/scripts/storage/directives/dtv-storage-selector.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        return {
+          restrict: 'EA',
+          scope: {
+            type: '@',
+            label: '@',
+            selected: '='
+          },
+          templateUrl: 'partials/storage/storage-selector.html',
+          link: function ($scope) {
+            var type = $scope.type ? $scope.type : 'single-file';
```
Comment (alex-deaconu) [inline web/scripts/storage/directives/dtv-storage-selector.js]: I was thinking to use a global variable for this. It is used everywhere:
https://github.com/Rise-Vision/rise-vision-apps/pull/109/files#diff-2b14aca859b1b6e81772d07da0982b9aR13

Suggestions? Thanks.

File: web/scripts/storage/filters/ftr-filters.js [hunk 3]
```diff
@@ -0,0 +1,130 @@
+'use strict';
+
+/* Filters */
+angular.module('risevision.storage.filters', ['risevision.common.i18n'])
```
Comment (alex-deaconu) [inline web/scripts/storage/filters/ftr-filters.js]: Copied this file and tests straight from Storage. Not sure which filters are actually used.

File: web/scripts/storage/services/svc-file-selector-factory.js [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+  .value('STORAGE_FILE_URL', 'https://storage.googleapis.com/')
+  .value('STORAGE_CLIENT_API', 'https://www.googleapis.com/storage/v1/b/')
+  .factory('fileSelectorFactory', ['$rootScope', '$window', 'filesFactory',
+    'userState', 'gadgetsApi',
+    '$loading', 'filterFilter', 'STORAGE_CLIENT_API', 'STORAGE_FILE_URL',
+    function ($rootScope, $window, filesFactory, userState, gadgetsApi,
+      $loading, filterFilter, STORAGE_CLIENT_API, STORAGE_FILE_URL) {
+      var factory = {};
+
+      factory.storageFull = false;
```
Comment (alex-deaconu) [inline web/scripts/storage/services/svc-file-selector-factory.js]: This variable determines whether it is running Storage Full or Storage Selector.

File: web/scripts/storage/services/svc-files-factory.js [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+            });
+          }
+          
+          return resp;
+        }
+
+        var params = {};
+        if (svc.folderPath) {
+          params.folderPath = decodeURIComponent(svc.folderPath);
+          svc.statusDetails.folder = params.folder;
```
Comment (ezequielc) [inline web/scripts/storage/services/svc-files-factory.js]: @alex-deaconu It seems that `params.folder` will alwaqys be undefined as `var params = {}` was defined 3 lines above. Is this intentional? Thanks

Comment (alex-deaconu) [inline web/scripts/storage/services/svc-files-factory.js]: @ezequielc good catch. Should be `params.folderPath`. Will fix. Thanks.

File: web/partials/storage/storage-modal.html [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+<div class="modal-body storage-full" id="storageSelectorModal">
+  <button type="button" id="closeButton" class="close" ng-click="dismiss()" aria-hidden="true">
+    <i class="fa fa-times"></i>
+  </button>
+
+	<div class="padding-large-horizontal padding-large-bottom">
+	<div class="app-header half-top">
+	  <div class="title-block">
+	    <!-- App Title -->
+	    <h1 id="title">Select a File</h1>
```
Comment (ezequielc) [inline web/partials/storage/storage-modal.html]: Just as a reminder, we should extract strings for i18n before merging this branch. Cheers

File: web/scripts/storage/controllers/ctr-storage-selector-modal.js (hunk 1 above)
Comment (ezequielc) [inline web/scripts/storage/controllers/ctr-storage-selector-modal.js]: Makes sense. Cheers

File: web/scripts/storage/directives/dtv-storage-selector.js [hunk 7]
```diff
@@ -0,0 +1,54 @@
+(function () {
+
+  'use strict';
+
+  // Declare legacy selector [error without it]
+  angular.module('risevision.widget.common.storage-selector', []);
```
Comment (ezequielc) [inline web/scripts/storage/directives/dtv-storage-selector.js]: @alex-deaconu Any ideas why we need to declare old selector module? Maybe for CH? Cheers

Comment (alex-deaconu) [inline web/scripts/storage/directives/dtv-storage-selector.js]: @ezequielc it's required in the `file-selector` which is used by the `background-component` which we use: https://github.com/Rise-Vision/widget-settings-ui-components/blob/master/src/_angular/file-selector/dtv-file-selector.js. Thanks.

File: web/scripts/storage/directives/dtv-storage-selector.js (hunk 2 above)
Comment (ezequielc) [inline web/scripts/storage/directives/dtv-storage-selector.js]: Yes, maybe we could have a dictionary object `SELECTOR_TYPES = {SINGLE_FILE: 'single-file' ...}` so we don't need to inject every type, specially when all are used like here: https://github.com/Rise-Vision/rise-vision-apps/pull/109/files#diff-2b14aca859b1b6e81772d07da0982b9aR48

File: web/scripts/storage/directives/dtv-storage-selector.js (hunk 7 above)
Comment (ezequielc) [inline web/scripts/storage/directives/dtv-storage-selector.js]: @alex-deaconu Thanks!

File: web/scripts/storage/services/svc-file-selector-factory.js [hunk 8]
```diff
@@ -0,0 +1,169 @@
...
+
+      factory.postFileToParent = function (file) {
+        var fileUrl = getFileUrl(file);
+        var data = {
+          params: fileUrl
+        };
+
+        if (!factory.storageFull) {
+          console.log('Message posted to parent window', [fileUrl]);
+          $window.parent.postMessage([fileUrl], '*');
```
Comment (ezequielc) [inline web/scripts/storage/services/svc-file-selector-factory.js]: @alex-deaconu Do we need this code for posting to parent? Thanks

**Comment by alex-deaconu** [comment]
@ezequielc I've fixed the typo and added a `SELECTOR_TYPES` object to be used throughout. See second commit. Thanks.

**Comment by ezequielc** [comment]
@alex-deaconu Thanks! Code looks good. I will start working on Add Folder. Cheers

**Comment by alex-deaconu** [comment]
@ezequielc rebased and squashed. Thanks.

**Comment by alex-deaconu** [comment]
Squashed. Ready for merge after review.

**Pull request merged**